numpy/core/src/multiarray/einsum_helpers.h

```diff
@@ -0,0 +1,52 @@
+#ifndef _NPY_EINSUM_HELPERS_H_
+#define _NPY_EINSUM_HELPERS_H_
+
+#define PY_SSIZE_T_CLEAN
+#include "Python.h"
+#include "structmember.h"
+
+#define NPY_NO_DEPRECATED_API NPY_API_VERSION
+#define _MULTIARRAYMODULE
+#include <numpy/npy_common.h>
+#include <numpy/arrayobject.h>
+#include <numpy/halffloat.h>
+#include <npy_pycompat.h>
+
+#include <ctype.h>
```

Comment:
I don't think it makes sense to fill this file with includes like this one. I've tried to address this in #17109.

Reply:
looks good to me, a private header shared between two sources, maybe I should rename it to `einsum_private.h`